app/models/alchemy/page/page_scopes.rb

```diff
@@ -36,7 +36,25 @@ module Page::PageScopes
 
       # All public pages
       #
-      scope :published, -> { where(public: true) }
+      scope :published, -> {
+        current_time = Time.current
+        where([
+          "#{Page.table_name}.public_on <= ? AND".freeze,
+          "(#{Page.table_name}.public_until IS NULL".freeze,
+          "OR #{Page.table_name}.public_until >= ?)".freeze
+        ].join(' '.freeze), current_time, current_time)
+      }
+
+      # All not public pages
+      #
+      scope :not_public, -> {
```

Comment:
Style/Lambda: Use the `lambda` method for multi-line lambdas. (https://github.com/bbatsov/ruby-style-guide#lambda-multi-line)